feelpp/feel/feelcore/exceptions.cpp

```diff
@@ -0,0 +1,92 @@
+ modify it under the terms of the GNU Lesser General Public
+ License as published by the Free Software Foundation; either
+ version 2.1 of the License, or (at your option) any later version.
+
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public
+ License along with this library; if not, write to the Free Software
+ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+#include <fmt/core.h>
+#include <feel/feelcore/environment.hpp>
+
+namespace Feel
+{
+
+void handleExceptions()
+{
+    try
+    {
+        throw; // re-throw exception already in flight } 
+    }
+    catch(const boost::filesystem::filesystem_error& e)
+    {
+        if ( Environment::isMasterRank() )
+        {
+            if ( e.code() == boost::system::errc::permission_denied )
+                fmt::print( "[feel++.boost.filesystem.filesystem_error.permission_denied] {}", e.what() );
+            else
+                fmt::print( "[feel++.boost.filesystem.filesystem_error] {}", e.what() );
+        }
+    }
+    catch (json::exception& e)
+    {
+        if ( Environment::isMasterRank() )
+            fmt::print("[feel++.json.exception] {}",e.what());
+    }
+    catch ( std::invalid_argument const& e )
+    {
+        Feel::cout << fmt::format( "[feelpp.std.invalid_argument] {}",e.what()) << std::endl;
+    }
+    catch (const std::runtime_error & e) 
+    {
+        if ( Environment::isMasterRank() )
+            fmt::print( "[feel++.std.runtime_error] {}", e.what() );
+    }
+    catch ( const std::exception& e )
+    {
+        if ( Environment::isMasterRank() )
+            fmt::print( "[feel++.std.exception] {}", e.what() );
+    }
+    if ( GitMetadata::populated() )
+    {
+        if ( GitMetadata::anyUncommittedChanges() )
+        {
+            std::cerr << fmt::format("WARN: there were uncommitted changes at build-time.") << std::endl;
+        }
+        const char* str = R"(commit {} (HEAD)
+                             describe {}
+                             Author: {} <{}>
+                             Date: {}
+                             Subject: {}
+                             Body: {}
+                            )";
+        std::cout << fmt::format( str,
+                        GitMetadata::commitSHA1(),
+                        GitMetadata::describe(),
+                        GitMetadata::authorName(),GitMetadata::authorEmail(),
+                        GitMetadata::commitDate(),
+                        GitMetadata::commitSubject(),
+                        GitMetadata::commitBody() );
+    }
+    else
+    {
+        std::cerr << "WARN: failed to get the current git state. Is this a git repo?" << std::endl;
+    }
```

Comment:
maybe call MPI_Abort, else we can have deadlock

Reply:
Indeed exception may be thrown by some MPI process, I will check this testcase

Reply:
done